tests/Database/DatabaseEloquentModelTest.php

```diff
@@ -1152,6 +1152,40 @@ public function testModelAttributesAreCastedWhenPresentInCastsArray()
 	}
 
 
+	public function testModelAttributeCastingPreservesNull()
```

Comment:
I happen to quite like it. In some cases you can get quite carried away, but in this case I think having it all in one method makes perfect sense.

All code in this method runs a test for one single line of code.